docs/data/data-grid/performance/performance.md

```diff
@@ -0,0 +1,47 @@
+# Data Grid - Performance
+
+<p class="description">Improve the performance of the DataGrid using the recommendations from this guide.</p>
+
+## Memoize inner components with `React.memo`
+
+The `DataGrid` component is composed of a central state object where all data is stored.
+When an API method is called, a prop changes, or the user interacts with the UI (e.g. filtering a column), this state object is updated with the changes made.
+To reflect the changes in the interface, the component must re-render.
+Since the state behaves like `React.useState`, it means that the `DataGrid` component will re-render as well as its children, which includes column headers, rows and cells.
+With smaller datasets, this is not a problem for concern, but it can become a bottleneck if the number of rows gets increased and, specially, if many columns render [custom content](/x/react-data-grid/column-definition/#rendering-cells).
+One way to overcome this issue is using `React.memo` to only re-render the children components when their props have changed.
+To start using memoization, import the inner components, then pass their memoized version to the respective slots, as follow:
+
+```tsx
+import {
+  GridRow,
+  DataGrid, // or DataGridPro, DataGridPremium
+  DataGridColumnHeaders, // or DataGridProColumnHeaders, DataGridPremiumColumnHeaders
```

Comment:
Side note: This seems inconsistent with the other components' names that do not change regardless of the package used (one exception is `DataGrid`/`DataGridPro`/`DataGridPremium`).
Should we rename it to `GridColumnHeaders` for all packages?